src/python/pants/core/util_rules/partitions.py

```diff
@@ -49,8 +49,15 @@ def description(self) -> str:
 PartitionElementT = TypeVar("PartitionElementT")
 
 
+@dataclass(frozen=True)
+@runtime_ignore_subscripts
+class Partition(Generic[PartitionKeyT, PartitionElementT]):
+    key: PartitionKeyT | None
+    elements: tuple[PartitionElementT, ...]
```

Comment:
Since the elements is required and the key is optional, can we swap these? Same for the type params.